xmppserver/src/main/java/org/jivesoftware/openfire/muc/spi/LocalMUCUser.java

```diff
@@ -587,16 +618,21 @@ private void process(
         @Nullable final MUCRole preExistingRole )
     {
         // Packets to a specific node/group/room
-        if ( preExistingRole == null )
+        boolean allowMemberListRetrival = false; 
```

Comment:
Before your changes, the code that follows this if-statement assumes that there is a non-null `preExistingRole`. This commit modifies that code in various places to allow for null values. Although it probably works fine, it will also be very error-prone. I believe that it would be an improvement to take the case where `preExistingRole` is null (basically any use-case where a non-occupant is trying to interact with the room and let that be processed by a different method completely. That way, we need not make the existing code more complex.